### DISCUSSION TITLE
How does PYG process duplicate edges?

### DISCUSSION THREAD

**Discussion by YangxinCs** [pyg-team/pytorch_geometric#7427] [Q&A] [answered]
Hi,Thank you for providing a convenient tool for deep graph learning.But I have some questions about the use of this tool.If 
I accidentally add duplicate edges when processing data, will the graph constructed by PYG become a complex graph instead of a simple graph? Are duplicate edges treated as parallel edges (multiple edges with the same edge between a pair of nodes)?

Looking forward to your answer.
YangXin

**Comment by songsong0425** [comment]
Hi. If you concern about the duplicated edges in the graph, you can simply run the `RemoveDuplicatedEdges` function in `torch_geometric.transforms`.

For example, here is a graph with duplicated edge:
```python
# edge between 0 and 1 is duplicated
edge_index = torch.tensor([[0, 1, 1, 2, 0],
                           [1, 0, 2, 1, 1]], dtype=torch.long)
x= torch.tensor([[-1], [0], [1]], dtype=torch.float)

data = Data(x=x, edge_index=edge_index)
data
# Data(x=[3, 1], edge_index=[2, 5])
```

You can get a unique edges after run `RemoveDuplicatedEdges()` as below:
```python
from torch_geometric.transforms import RemoveDuplicatedEdges

transform = RemoveDuplicatedEdges()
out = transform(data)
out
# Data(x=[3, 1], edge_index=[2, 4])
```

References: \
https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.transforms.RemoveDuplicatedEdges.html#torch_geometric.transforms.RemoveDuplicatedEdges
https://github.com/pyg-team/pytorch_geometric/blob/master/test/transforms/test_remove_duplicated_edges.py

**Reply by YangxinCs** [reply]
thank you very much!

**Comment by rusty1s** [comment, marked as answer]
> Are duplicate edges treated as parallel edges (multiple edges with the same edge between a pair of nodes)

Yes :)

**Reply by YangxinCs** [reply]
thank you very much!
